.github/workflows/spack-deployment-workflow.yml

```diff
@@ -2,39 +2,80 @@
-      - name: Set output
+      - name: Get release tag
         id: vars
-        run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
-      - name: Check output
-        env:
-          RELEASE_VERSION: ${{ steps.vars.outputs.tag }}
         run: |
-          echo $RELEASE_VERSION
-          echo ${{ steps.vars.outputs.tag }}
-      - uses: docker/setup-buildx-action@v1
-        id: buildx
-        with:
-          install: true
-      - name: Build
+          GEOSCHEM_VERSION=${GITHUB_REF#refs/*/}
+          echo ::set-output name=release_tag::${GEOSCHEM_VERSION}
+          cd ${GITHUB_WORKSPACE}
+          git checkout $GEOSCHEM_VERSION
+          echo ::set-output name=release_sha::$(git rev-list -n 1 $GEOSCHEM_VERSION)
+      - name: Install dependencies
+        run: |
+          set -x
+          conda install -y -c conda-forge gh flake8 mypy isort
+      - name: Login with PAT
+        run: |
+          git config --global user.email "liam.bindle@gmail.com"
+          git config --global user.name "LiamBindle"
+          git config --global credential.helper store
+          echo "https://LiamBindle:${{ secrets.GH_PUSH_PAT }}@github.com/LiamBindle/spack.git" > ~/.git-credentials
+          echo ${{ secrets.GH_PUSH_PAT }} | gh auth login --with-token
+      - name: Update GCHP spack package and submit PR
         env:
-          RELEASE_VERSION: ${{ steps.vars.outputs.tag }}
+          GEOSCHEM_VERSION: ${{ steps.vars.outputs.release_tag }}
+          VERSION_STRING: version('${{ steps.vars.outputs.release_tag }}', commit='${{ steps.vars.outputs.release_sha }}',  submodules=True)
+          TAR_URL_PREFIX: https://github.com/geoschem/GCHP/archive
         run: |
-          docker build -f ./.github/workflows/spack-deployment/spack-deployment.dockerfile --build-arg gh_access_token=${{ secrets.GH_ACCESS_TOKEN }} --build-arg version_tag=$RELEASE_VERSION ./.github/workflows/spack-deployment/
+          set -x
+          TEMP_DIR=$(mktemp -d)
+          SPACK_DIR=$TEMP_DIR/spack
+          
+          # Checkout spack/spack:develop
+          git clone https://github.com/spack/spack.git $SPACK_DIR
+          cd $SPACK_DIR
+          git checkout develop
+          . share/spack/setup-env.sh
+
+          # Add LiamBindle/spack as a remote (PR will come from this repo)
+          git remote add pr-source-repo https://github.com/LiamBindle/spack.git
+          
+          # Create a new branch for current release
+          git checkout -b gchp-${GEOSCHEM_VERSION}
+
+          # replace the tarball link to new version
+          sed -i "s|https://github.com/geoschem/GCHP/archive/.*.gz|${TAR_URL_PREFIX}/${GEOSCHEM_VERSION}.tar.gz|" var/spack/repos/builtin/packages/gchp/package.py
+          # add new line to version history
+          # todo: uncomment (needs fix) | sed -i "1,/version(.*/s//${VERSION_STRING}\n    &/" var/spack/repos/builtin/packages/gchp/package.py
+
+          # test that style is up to snuff
+          spack style
```

Comment:
I was having issues with spack if I didn't install a bunch of dependencies for spack. I'm impressed that spack style is working correctly. Did you run into any errors when installing spack?

Reply:
I didn't but I was having trouble with the alpine tag, so maybe it's a bit unstable. I had to install `isort` but that was the only extra I needed.